src/rpc/blockchain.cpp

```diff
@@ -1640,6 +1640,37 @@ static T CalculateTruncatedMedian(std::vector<T>& scores)
     }
 }
 
+static constexpr int NUM_GETBLOCKSTATS_PERCENTILES = 5;
+
+template<typename T>
+static void CalculatePercentilesByWeight(T result[NUM_GETBLOCKSTATS_PERCENTILES], std::vector<std::pair<T, int64_t>>& scores, int64_t total_weight)
+{
+    const size_t size = scores.size();
+    if (size == 0) {
+        return;
+    }
+
+    std::sort(scores.begin(), scores.end());
+
+    // 10th, 25th, 50th, 75th, and 90th percentile weight units.
+    const double weights[NUM_GETBLOCKSTATS_PERCENTILES] = {
+        total_weight / 10.0, total_weight / 4.0, total_weight / 2.0, (total_weight * 3.0) / 4.0, (total_weight * 9.0) / 10.0
+    };
+    int64_t next_percentile_index = 0;
+    int64_t cumulative_weight = 0;
+    for (const auto& element: scores) {
+        cumulative_weight += element.second;
+        while (cumulative_weight >= weights[next_percentile_index]) {
```

Comment:
This should be `while (next_percentile_index < NUM_GETBLOCKSTATS_PERCENTILS && cumulative_weight >= weights[next_percentile_index]) {`, no?

Reply:
Yes, thanks for catching that.